DOC-5131 first attempt at links/formatting for develop landing page

andy-stark-redis opened the pull request:
[DOC-5131](https://redislabs.atlassian.net/browse/DOC-5131)

I've added a table for the client and tool links and thrown in a few speculative formatting ideas. Also, I've tried removing the page ToC bar from the right-hand side.

BTW, this also fixes something I didn't notice on the previous PR - all section index pages using the list.html layout had the CLI widget at the top (ie, index pages for data types, client APIs, etc). Maybe we should leave that in, actually - I don't know.

Anyway, all (polite) feedback welcome :-) 

[DOC-5131]: https://redislabs.atlassian.net/browse/DOC-5131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dwdougherty reviewed (changes requested):
1. I'm always polite. 🙂
2. **I like the content**; I think it's just the right amount. Michelle might have other ideas though. 😉
3. Rather than using an entirely new layout to exclude the right side TOC, I'm wondering if you could instead use a bit of frontmatter, say `hideTOC: [true | false]` to show/hide the TOC as appropriate for each `_index.md` page. It would work just like the `hideListLinks` frontmatter. It would also be easier for other folks to use that way, though they would have to make the same changes to the `operate` list layout.

```
{{ if not .Params.hideTOC }}
      {{ partial "docs-toc.html" . }}
{{ end }}
```

4. The images don't show in preview (but they do locally). You might consider using the `image` shortcode; it's used very widely in the K8s, RC, and RS docs.
5. If you don't intend on reusing the content elsewhere, I think it's better to "inline" the content currently stored in the two shortcodes. Or maybe consider using embeds.

Feel free to reach out via Slack if you want to discuss.

dwdougherty reviewed (approved):
You can leave the content shortcodes for now. As for the rest, it looks good. I'll go ahead and approve, but please don't merge until @mich-elle-luna weighs in.

andy-stark-redis commented:
> It's weird that it doesn't show a merge conflict. As it stands now, were you to merge this PR, it would overwrite a change I made last night to the develop list.html layout.

@dwdougherty I resolved the merge conflict earlier.

dwdougherty commented:
The file still shows the old test:

 ```
<!-- Responsive CLI Display Block as Table -->
 {{ if eq .RelPermalink "/develop/" }}
```

The test should be:

`{{ if strings.HasSuffix .RelPermalink "/develop/" }}`

dwdougherty commented on layouts/develop/list.html:
```diff
@@ -36,9 +36,9 @@ <h1>
+        {{ if strings.HasSuffix .RelPermalink "/develop/" }}
         <div class="overflow-x-auto">
-          <table class="w-full border-collapse md:table">
+          <table class="w-full border-collapse md:table" style="background-color: rgb(240,240,240)">
```
Please change this line to the following to stick with Tailwind CSS styling.

`<table class="w-full border-collapse md:table bg-neutral-150">`

dwdougherty commented on layouts/shortcodes/dev-features.html:
```diff
@@ -0,0 +1,55 @@
+    </tr>
+    <tr>
+        <td>
+            <ul style="list-style-type: none; padding-left:0%;margin-left:-2%">
```
Similar to my previous comment, here's the equivalent styling for the above:

`<ul class="list-none pl-0 -ml-[2%]">`

dwdougherty replied:
Please replace styling with this Tailwind CSS class:

`class="list-none pl-0 -ml-[2%]"`

Same for the other instances further down.

mich-elle-luna commented:
I think this is looking really nice! I might suggest that the screenshots for insight and vscode appear in a larger size to break up the page a bit more and highlight those UIs, but I think this is a great update to start with and see how it goes. Thank you!

mich-elle-luna reviewed (approved):
thank you!

dwdougherty commented:
One final comment: Do you know why the three columns of the last table are not spread out evenly on the page? This is triggering my OCD! 😂

mich-elle-luna commented:
I'm not sure if the table column width issue is connected, but the footer should appear all the way across the page. Right now, it is stuck on the right side and needs to be fixed before merging.

dwdougherty commented:
At first, I thought it was the hiding of the right side TOC that made a mess of the footer. Then, on a whim, I decided to replace the HTML tables with inline Markdown tables and that made the footer issue go away. No idea why.

@andy-stark-redis Here's the [`_index.md`](https://github.com/user-attachments/files/20259400/_index.md) file I was working on for you to take a look at (attachment).

dwdougherty commented on layouts/shortcodes/client-splash.html:
```diff
@@ -0,0 +1,111 @@
+<table>
+<tr>
+<td>
```
Please remove these three lines, which cause the footer to behave erratically.

dwdougherty commented:
The first three lines of the client-splash.html file were redundant. See code comment above. This is what was fouling the footer.

andy-stark-redis commented:
@dwdougherty Thanks for your new `_index.md` content - it fixes all the issues and avoids any remaining need for the two shortcode files, so I've added it now (and deleted the shortcode files).

@mich-elle-luna David's updates fix the footer issue now and also the two screenshots are now a bit bigger. See what you think.

dwdougherty reviewed (approved):
LG(reat)TM!

dwdougherty commented:
I'm not crying! You're crying! 🤣

mich-elle-luna reviewed (approved):
thank you!
